File: C++/717. 1-bit and 2-bit Characters.cpp
class Solution {
public:
    bool isOneBitCharacter(vector<int>& bits) {
        int len = bits.size();
        int ans = true;
        for(int i=0;i<len;)
        {
            if(bits[i])
            {
                ans = false;
                i += 2;
            }
            else
            {
                ans = true;
                i += 1;
            }
        }
        return ans;
    }
};

File: Python/717. 1-bit and 2-bit Characters.py
class Solution:
    def isOneBitCharacter(self, bits):
        """
        :type bits: List[int]
        :rtype: bool
        """
        ans, i = True, 0
        while i < len(bits):
            if bits[i] == 1:
                i += 2
                ans = False
            else:
                i += 1
                ans = True
        return ans